tiff_2_png/tiff_2_png_converter.py

```diff
@@ -228,21 +241,34 @@ def convert_to_png(
         logging.info(f"MIN Value in volume: {global_min_pixel}")
 
         # convert to 8 bit
-        images_in_sequence_8_bit = [None] * num_files_in_sequence
+        images_in_sequence_n_bits = [None] * num_files_in_sequence
+            
+        label_bit_text = ("16","8")[scale_to_8_bit == True] 
+        
         with typer.progressbar(
-            range(num_files_in_sequence), label="Rescaling to 8 bit signal"
+            range(num_files_in_sequence), label=f"Rescaling to {label_bit_text} bit signal"
         ) as progress:
             for slice in progress:
-                _img = images_in_sequence[slice]
-                img_8_bit = np.subtract(_img, global_min_pixel)
-                img_8_bit = np.true_divide(_img, global_max_pixel - global_min_pixel)
-                img_8_bit = np.multiply(img_8_bit, 255).astype(np.uint8)
-                if gamma is not None:
-                    img_8_bit = iph.gamma_correction(img_8_bit, gamma)
-                images_in_sequence_8_bit[slice] = img_8_bit
+                
+                if bits_per_sample == np.uint16:
+                    img_16_bit = images_in_sequence[slice]
+                    if scale_to_8_bit:
+                        global_scale = float(global_max_pixel - global_min_pixel)
+                        scale = 255.0 / global_scale
+                        byte_data = (img_16_bit - global_min_pixel) * scale 
+                        img_8_bit = (byte_data.clip(0, 255) + 0.5).astype(np.uint8)
+                        images_in_sequence_n_bits[slice] = img_8_bit
+                    else:
+                        images_in_sequence_n_bits[slice] = img_16_bit
+                else:
+                    img_8_bit = images_in_sequence[slice]
+                    if gamma is not None:
```

Comment:
I don't think the `is` is needed.
```suggestion
                    if gamma not None:
```

Reply:
It doesnt compile on my python. I reversed it to `if gamma is not None:`